harness/determined/cli/shell.py

```diff
@@ -36,46 +33,44 @@ def start_shell(args: Namespace) -> None:
     config = parse_config(args.config_file, None, args.config, args.volume)
     resp = launch_command(
         args.master,
-        "shells",
+        "api/v1/shells",
         config,
         args.template,
         context_path=args.context,
         data=data,
-    )
+    )["shell"]
+    print(resp)
 
     if args.detach:
         print(resp["id"])
         return
 
-    command = None
+    ready = False
```

Comment:
This logic below seems a bit complicated to me with the condition variable you are setting here. Why do you need to break out of the `with api.ws(args.master, "shells/{}/events".format(resp["id"])) as ws:` before calling `_open_shell`?

Reply:
I didn't change the logic of using the condition variable. I just removed the use of the `snapshot` field, which belongs to the old API. The logic is waiting for the state of a shell to change to running. It used to get the shell id from the websocket event. It is just changed to make a different call to the master.